projects/baby_driver/scripts/test_spi_exchange.py

```diff
@@ -0,0 +1,141 @@
+"""This Modules tests methods in spi_exchange.py"""
+
+import unittest
+from unittest.mock import patch
+
+from spi_exchange import spi_exchange
+from gpio_port import GpioPort
+from can_util import Message
+from message_defs import BABYDRIVER_DEVICE_ID, BABYDRIVER_CAN_MESSAGE_ID, BabydriverMessageId
+
+OK_STATUS = 0
+FAILING_STATUS = 1
+class TestSPIExchange(unittest.TestCase):
+    """Test spi_exchange function"""
+
+    @patch('can_util.send_message')
+    @patch('can_util.next_message')
+    def test_send_message(self, mock_next_message, mock_send_message):
```

Comment:
Your tests are fairly comprehensive, I would also like to see tests with `len(tx_bytes) == 255` and with `rx_len == 255` since they're the maximum values. You can use Python's multiplying lists by ints feature to help with constructing the necessary lists `[0] * 5 == [0, 0, 0, 0, 0]`.